frontend/src/app/questions/questions.component.html

```diff
@@ -0,0 +1,158 @@
+    <p-toast />
+    <ng-container *ngIf="!loading">
+        <p-toolbar styleClass="mb-2 gap-2">
+            <div class="p-justify-end">
+                <p-button
+                    icon="pi pi-plus"
+                    severity="primary"
+                    [outlined]="true"
+                    label="New"
+                    class="mr-2"
+                    (onClick)="openNewQuestion()" />
+                <p-button
+                    icon="pi pi-trash"
+                    severity="danger"
+                    label="Delete"
+                    (onClick)="deleteSelectedQuestions()"
+                    [disabled]="!selectedQuestions || !selectedQuestions.length" />
+            </div>
+        </p-toolbar>
+        <p-table
+            sortField="id"
+            [sortOrder]="1"
+            [columns]="cols"
+            [value]="questions"
+            [(selection)]="selectedQuestions"
+            datakey="id"
+            [tableStyle]="{ 'table-layout': 'fixed' }"
+            [paginator]="true"
+            [rows]="5"
+            [rowsPerPageOptions]="[5, 10, 20]"
+            styleClass="p-datatable-gridlines-striped">
+            <ng-template pTemplate="caption">
+                <div class="flex">
+                    <h3 class="m-0">Manage Questions</h3>
+                </div>
+            </ng-template>
+            <ng-template pTemplate="header" let-columns>
+                <tr>
+                    <th style="width: 10%"><p-tableHeaderCheckbox /></th>
+                    <th pSortableColumn="id" style="width: 10%">Id <p-sortIcon field="id" /></th>
+                    <th style="width: 10%">Title</th>
+                    <th style="width: 40%">Description</th>
+                    <th style="width: 10%">Topics</th>
+                    <th style="width: 10%">Difficulty</th>
+                    <th style="width: 10%"></th>
+                </tr>
+            </ng-template>
+            <ng-template pTemplate="body" let-question>
+                <tr>
+                    <td>
+                        <p-tableCheckbox [value]="question" />
+                    </td>
+                    <td>{{ question.id }}</td>
+                    <td>{{ question.title }}</td>
+                    <td>{{ question.description }}</td>
+                    <td>{{ question.topics.join(', ') }}</td>
+                    <td>{{ question.difficulty }}</td>
+                    <td>
+                        <p-button
+                            label="Edit"
+                            severity="primary"
+                            icon="pi pi-file-edit"
+                            class="mr-2"
+                            [text]="true"
+                            (onClick)="editQuestion(question)" />
+                    </td>
+                </tr>
+            </ng-template>
+        </p-table>
+    </ng-container>
+    <p-dialog header="Header" [(visible)]="isDialogVisible" [modal]="true" [style]="{ width: '25rem' }">
```

Comment:
I think it would be a lot cleaner to move the dialog to be a separate component. It could be something like this:
```typescript
export class QuestionDialogComponent implements OnInit {
    @Input() questionData!: Question;
    @Output() dialogClosed: EventEmitter<void> = new EventEmitter();
    protected isEditMode: boolean = false;

    ngOnInit(): void {
        if (this.questionData) {
            this.isEditMode = true;
            // Set the form data
        }
    }

    // TODO   
}
```